internal/loader/configuration_setting_loader.go

```diff
@@ -535,26 +538,48 @@ func getConfigurationSettings(
 	nullString := "\x00"
 
 	for _, filter := range filters {
-		if filter.LabelFilter == nil {
-			filter.LabelFilter = &nullString // NUL is escaped to \x00 in golang
-		}
-		selector := azappconfig.SettingSelector{
-			KeyFilter:   &filter.KeyFilter,
-			LabelFilter: filter.LabelFilter,
-			Fields:      azappconfig.AllSettingFields(),
-		}
-		pager := client.NewListSettingsPager(selector, nil)
+		if filter.KeyFilter != nil {
+			if filter.LabelFilter == nil {
+				filter.LabelFilter = &nullString // NUL is escaped to \x00 in golang
+			}
+			selector := azappconfig.SettingSelector{
+				KeyFilter:   filter.KeyFilter,
+				LabelFilter: filter.LabelFilter,
+				Fields:      azappconfig.AllSettingFields(),
+			}
+			pager := client.NewListSettingsPager(selector, nil)
 
-		for pager.More() {
-			page, err := pager.NextPage(ctx)
+			for pager.More() {
+				page, err := pager.NextPage(ctx)
+				if err != nil {
+					e <- err
+				} else if len(page.Settings) > 0 {
+					c <- page.Settings
+				}
+			}
+		} else {
+			snapshot, err := client.GetSnapshot(ctx, *filter.SnapshotName, nil)
 			if err != nil {
 				e <- err
-			} else if len(page.Settings) > 0 {
-				c <- page.Settings
 			}
-		}
 
+			if *snapshot.CompositionType != azappconfig.CompositionTypeKey {
```

Comment:
why only accept 'key' type and not allowing key+lable?

Reply:
This is by design, it's consistent with other providers. support key+label type would cause ambiguity problem, can't which label would be populated is not definitive